snaps/learn/tutorials/transaction-insights.md

```diff
@@ -32,70 +38,493 @@ starter kit by running:
+into that `ìmages` folder.  
+
+<details>
+  <summary>Icon attribution</summary>
+  <div>
+This is a free icon, "Gas" by Mello from the [Noun Project](https://thenounproject.com/browse/icons/term/gas/).
+ </div>
+</details>
+
+
+Your file structure should look like this:
+
+```text
+transaction-insights-snap/
+├─ packages/
+│  ├─ site/
+|  |  |- src/
+|  |  |  |- App.tsx
+|  |  ├─ package.json
+|  |  |- ...(React app content)
+|  |
+│  ├─ snap/
+|  |  ├─ images/
+|  |  |  |- gas.svg
+|  |  ├─ src/
+|  |  |  |- index.test.ts
+|  |  |  |- index.ts
+|  |  ├─ snap.manifest.json
+|  |  ├─ package.json
+|  |  |- ... (Snap content)
+├─ package.json
+├─ ... (other stuff)
+```
 
-```bash
-yarn run allow-scripts auto
+Open `packages/snap/snap.manifest.json` in a text editor. This file contains the main configuration 
+details for your Snap. Edit the `npm` object, within the `location` object, and add `iconPath` with 
+the value `"images/gas.svg"` to point to your new icon:
+
+<Tabs>
+  <TabItem value="Line updates" label="Update snap.manifest.json" default>
+
+```json title="snap.manifest.json"
+"location": {
+  "npm": {
+    "filePath": "dist/bundle.js",
+    "iconPath": "images/gas.svg",
+    "packageName": "snap",
+    "registry": "https://registry.npmjs.org/"
+  }
 ```
+</TabItem>
+<TabItem value="Full page" label="Example" default>
 
-### 2. Enable transaction insights and the Ethereum provider
+```json title="snap.manifest.json"
+{
+  "version": "0.1.0",
+  "description": "An example Snap written in TypeScript.",
+  "proposedName": "TypeScript Example",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/MetaMask/template-snap-monorepo.git"
+  },
+  "source": {
+    "shasum": "TwVBeJ/PL4uARBEFZmgJukYd3XWyTbilkOPhtdg+GvU=",
+    "location": {
+      "npm": {
+        "filePath": "dist/bundle.js",
+        "iconPath": "images/gas.svg",
+        "packageName": "snap",
+        "registry": "https://registry.npmjs.org/"
+      }
+    }
+  },
+   "initialPermissions": {
+    "snap_dialog": {},
+    "endowment:rpc": {
+      "dapps": true,
+      "snaps": false
+    },
+    "endowment:network-access": {}
+  },
+  "manifestVersion": "0.1"
+}
+```
+</TabItem>
+</Tabs>
 
-The default template Snap, such as the one in
-[Create a gas estimation Snap](gas-estimation.md),
-is configured to expose a JSON-RPC API with a simple hello command, which brings up a dialog box.
-In contrast, the Snap you're creating in this tutorial doesn't expose any API.
-Instead, it provides transaction insights directly in MetaMask's transaction confirmation window.
+Open `packages/snap/package.json` in a text editor. Edit the `files` array and reference the 
+`images/` folder:
+
+<Tabs>
+  <TabItem value="Line updates" label="Update package.json" default>
+
+```json title="package.json"
+"files": [
+  "dist/",
+  "images/",
+  "snap.manifest.json"
+],
+```
+
+</TabItem>
+<TabItem value="Full page" label="Example" default>
```

Comment:
Same comment as above, I think we can remove this example tab.

Reply:
Sure -- would love some user testing to know how low our entry-level is, but will elipse code fragments instead